### PR TITLE
Queue with array

### PR DESCRIPTION
**Pull request by SmrutiSuman**

### CODE REVIEW

File: queue/queue.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    @front = @rear = -1
+    @queue = []
+    @size = size
+  end
+
+  def enqueue_item(item)
+    if @queue.empty?
+      @rear = @front = 0
+    elsif @size == @rear
+      raise "Cann't insert element because Queue is full."
```
[the remark this answers is not in the document]
Comment (SmrutiSuman) [reply queue/queue.rb]: Done

File: queue/tc_queue.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    assert_equal([], que.queue)
+  end
+
+  def test_search_item
+    queue = Queue.new(10)
+    array_list = [2,3,4,51,1,34]
+    array_list.each do |i|
+      queue.enqueue_item(i)
+    end
+    assert_includes(true, queue.search_item(4))
```
Comment (Sushanthbc) [inline queue/tc_queue.rb]: `asset_equal` makes more sense

File: queue/tc_queue.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  def test_add_item_to_queue
+    que = Queue.new(10)
+    que.enqueue_item(8)
+    assert_equal([8], que.queue)
+  end
+
+  def test_remove_item
+    que = Queue.new(10)
+    que.enqueue_item(8)
+    que.dequeue_item(8)
```
Comment (Sushanthbc) [inline queue/tc_queue.rb]: Just use `que.dequeue_item`

File: queue/queue.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    else
+      @rear += 1
+    end
+    @queue << item
+  end
+
+  def dequeue_item
+    if @queue.empty?
+      raise "Queue is empty"
```
[the remark this answers is not in the document]
Comment (SmrutiSuman) [reply queue/queue.rb]: Done

File: stack/stack.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+class Stack
```
Comment (Sushanthbc) [inline stack/stack.rb]: How is stacks showing up here, this will cause merge conflicts.

**Pull request merged**